pkg/vault/contracts/RouterHooks.sol

```diff
@@ -14,24 +14,27 @@ import "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
 import "@balancer-labs/v3-interfaces/contracts/vault/RouterTypes.sol";
 
 import { RouterWethLib } from "./lib/RouterWethLib.sol";
-import { RouterQueries } from "./RouterQueries.sol";
+import { RouterCommon } from "./RouterCommon.sol";
 
 /**
  * @notice Base Router with hooks for swaps and liquidity operations via Vault.
  * @dev Implements hooks for init, add liquidity, remove liquidity, and swaps.
  */
-contract BaseRouter is RouterQueries {
+abstract contract RouterHooks is RouterCommon {
     using Address for address payable;
     using RouterWethLib for IWETH;
     using SafeCast for *;
 
-    constructor(
-        IVault vault,
-        IWETH weth,
-        IPermit2 permit2,
-        string memory routerVersion
-    ) RouterQueries(vault, weth, permit2, routerVersion) {
-        // solhint-disable-previous-line no-empty-blocks
+    /**
+     * @notice The sender does not transfer the correct amount of tokens to the Vault.
+     * @param token The address of the token that was expected to be transferred.
+     */
+    error InsufficientPayment(IERC20 token);
+
+    bool internal immutable _isAggregator;
+
+    constructor(bool isAggregator) {
+        _isAggregator = isAggregator;
```

Comment:
Sorry for being annoying, but I'd much rather leave everything related with this flag for another PR.

The current `Router` only knows how to settle with permit2 in the main branch, and that should stay the same in this PR. This PR was meant to be a refactor only so that we can focus on restructuring the code specifically, without adding anything. The diff is already quite big, and I want to change a few things around settlement when `isAggregator` is true. But we'll be able to discuss that more efficiently in a different scope.

Can we keep the router's features the same for the time being?